code/game/objects/items/holy_weapons.dm

```diff
@@ -442,6 +442,129 @@
+		swordright.armour_penetration = armour_penetration
+		swordright.block_chance = block_chance
+		swordright.wound_bonus = wound_bonus
+		swordright.bare_wound_bonus = bare_wound_bonus
+		swordright.sheath = src
+		user.put_in_r_hand(swordright)
+
+		swordleft = new(src)
+		swordleft.force = force
+		swordleft.armour_penetration = armour_penetration
+		swordleft.block_chance = block_chance
+		swordleft.wound_bonus = wound_bonus
+		swordleft.bare_wound_bonus = bare_wound_bonus
+		swordleft.sheath = src
+		user.put_in_l_hand(swordleft)
+
+		to_chat(user, "you draw your righteous blades.")
+		playsound(user, 'sound/items/unsheath.ogg', 25, TRUE)
+		swords = FALSE
+		update_icon()
+
+/obj/item/nullrod/dualsword/attackby(obj/item/I, mob/living/user, params)
+	. = ..()
+	if(istype(I, /obj/item/nullrod/handedsword))
+		qdel(I)
+
+		var/otherhand = user.get_inactive_held_item()
+		if(istype(otherhand, /obj/item/nullrod/handedsword))
+			qdel(otherhand)
+
+		to_chat(user, "you sheathe your blades.")
+		playsound(user, 'sound/items/sheath.ogg', 25, TRUE)
+		swords = TRUE
+		update_icon()
+		
+/obj/item/nullrod/dualsword/update_icon()
+	. = ..()
+	item_state = swords ? "fulldual" : "emptydual"
+	icon_state = item_state
+	if(ishuman(loc))
+		var/mob/living/carbon/human/H = loc
+		H.update_inv_belt()
+
+/obj/item/nullrod/handedsword
+	name = "Justice"
+	desc = "Ashes to ashes... Rust to rust..."
+	icon = 'icons/obj/weapons/swords.dmi'
+	icon_state = "dualright"
+	item_state = "dualright"
+	lefthand_file = 'icons/mob/inhands/weapons/swords_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/weapons/swords_righthand.dmi'
+	attack_verb = list("attacked", "slashed", "stabbed", "sliced", "ripped", "diced", "cut")
+	hitsound = 'sound/weapons/rapierhit.ogg'
+	w_class = WEIGHT_CLASS_HUGE
+	chaplain_spawnable = FALSE
+	var/obj/item/nullrod/dualsword/sheath //so the sheathe is refilled when the swords are dropped
+
+/obj/item/nullrod/handedsword/other
+	name = "Splendor"
+	desc = "\"I'm going to ultrakill you!\" -Righteous Judge"
+	icon = 'icons/obj/weapons/swords.dmi'
+	icon_state = "dualleft"
+	item_state = "dualleft"
+
+/obj/item/nullrod/handedsword/attack(mob/living/M, mob/living/user, secondattack = FALSE)
+	. = ..()
+	var/obj/item/nullrod/handedsword/secondsword = user.get_inactive_held_item()
+	if(istype(secondsword, /obj/item/nullrod/handedsword) && !secondattack)
+		sleep(0.2 SECONDS)
+		if(QDELETED(secondsword) || QDELETED(src))
+			return
+		user.swap_hand()
+		secondsword.attack(M, user, TRUE)
+		user.changeNext_move(CLICK_CD_MELEE)
+	return
+
+/obj/item/nullrod/handedsword/dropped(mob/user, silent = TRUE)
+	. = ..()
+	if(QDELETED(src))
+		return
+	var/otherhand = user.get_inactive_held_item()
+	if(istype(otherhand, /obj/item/nullrod/handedsword))
+		qdel(otherhand)
+	if(sheath)
+		to_chat("you sheathe your blades.")
+		sheath.swords = TRUE
+		sheath.update_icon()
+		playsound(user, 'sound/items/sheath.ogg', 25, TRUE)
+	qdel(src)
```

Comment:
use forcemove, we don't want swords to restore integrity everytime they move to the bay + prevents exploits

Reply:
or maybe you do but still use something else